File: inetpub/wwwroot/AutohausPortal/lib/InfoCenterData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Data;
using CKG.Base.Kernel.Security;
using CKG.Base.Business;
using System.Data.SqlClient;

namespace AutohausPortal.lib
{
    public class InfoCenterData : BankBase
    {
        #region Declarations

        private DataTable m_tblDocumentTypes = new DataTable();
        private DataTable m_tblDocuments = new DataTable();
        private DataTable m_tblDocumentRights = new DataTable();
        private string m_uploadfile;

        #endregion

        #region Properties

        public DataTable DocumentTypes
        {
            get { return m_tblDocumentTypes; }
            set { m_tblDocumentTypes = value; }
        }

        public DataTable Documents
        {
            get { return m_tblDocuments; }
            set { m_tblDocuments = value; }
        }

        public DataTable DocumentRights
        {
            get { return m_tblDocumentRights; }
            set { m_tblDocumentRights = value; }
        }

        public string UploadFile
        {
            get { return m_uploadfile; }
            set { m_uploadfile = value; }
        }

        #endregion

        #region Methods

        public InfoCenterData(ref User objUser, ref App objApp, string strAppID, string strSessionID, string strFileName)
            :base(ref objUser, ref objApp, strAppID, strSessionID, strFileName)
        {
        }

        public void GetDocumentTypes()
        {
            using (SqlConnection cn = new SqlConnection(m_objApp.Connectionstring))
            {
                cn.Open();

                m_tblDocumentTypes.Clear();

                SqlDataAdapter daDocTypes = new SqlDataAdapter("SELECT * FROM DocumentType WHERE CustomerID = @CustomerID OR DocumentTypeId = @DocumentTypeId", cn);
                daDocTypes.SelectCommand.Parameters.AddWithValue("@CustomerID", m_objUser.Customer.CustomerId);
                daDocTypes.SelectCommand.Parameters.AddWithValue("@DocumentTypeId", 1);
                daDocTypes.Fill(m_tblDocumentTypes);

                cn.Close();
            }
        }

        /// <summary>
        /// Gibt alle Dokumente für eine Gruppe oder alle Dokumente des Kunden zurück
        /// </summary>
        /// <param name="groupId"></param>
        public void GetDocuments(int groupId = -1)
        {
            SqlDataAdapter daDocs;

            using (SqlConnection cn = new SqlConnection(m_objApp.Connectionstring))
            {
                cn.Open();

                m_tblDocuments.Clear();

                if (groupId == -1)
                {
                    daDocs = new SqlDataAdapter("SELECT * FROM vwDocumentWithType WHERE customerId = @customerId", cn);
                    daDocs.SelectCommand.Parameters.AddWithValue("@customerId", m_objUser.Customer.CustomerId);
                }
                else
                {
                    daDocs = new SqlDataAdapter("SELECT * FROM vwDocumentGroup WHERE groupId = @groupId", cn);
                    daDocs.SelectCommand.Parameters.AddWithValue("@groupId", groupId);
                }
                
                daDocs.Fill(m_tblDocuments);

                cn.Close();
            }
        }

        public void GetDocumentRights(int documentId)
        {
            using (SqlConnection cn = new SqlConnection(m_objApp.Connectionstring))
            {
                cn.Open();

                m_tblDocumentRights.Clear();

                SqlDataAdapter daDocs = new SqlDataAdapter("SELECT * FROM documentRights WHERE documentId = @documentId", cn);
                daDocs.SelectCommand.Parameters.AddWithValue("@documentId", documentId);
                daDocs.Fill(m_tblDocumentRights);

                cn.Close();
            }
        }

        public void SaveDocumentType(int docTypeId, string docType)
        {
            DataRow tablerow;
            DataRow[] rows;

            using (SqlConnection cn = new SqlConnection(m_objApp.Connectionstring))
            {
                cn.Open();

                SqlCommand cmd = new SqlCommand();
                cmd.Connection = cn;
                cmd.CommandType = CommandType.Text;

                if (docTypeId == -1)
                {
                    cmd.CommandText = "INSERT INTO DocumentType (docTypeName,customerId) VALUES (@docTypeName,@customerId);SELECT SCOPE_IDENTITY() AS 'Identity'";
                    cmd.Parameters.AddWithValue("@docTypeName", docType);
                    cmd.Parameters.AddWithValue("@customerId", m_objUser.Customer.CustomerId);
                    int newId = Convert.ToInt32(cmd.ExecuteScalar());
                    tablerow = m_tblDocumentTypes.NewRow();
                    tablerow["documentTypeId"] = newId;
                    tablerow["docTypeName"] = docType;
                    tablerow["customerId"] = m_objUser.Customer.CustomerId;
                    m_tblDocumentTypes.Rows.Add(tablerow);
                }
                else
                {
                    cmd.CommandText = "UPDATE DocumentType SET docTypeName = @docTypeName WHERE documentTypeId = @documentTypeId;";
                    cmd.Parameters.AddWithValue("@docTypeName", docType);
                    cmd.Parameters.AddWithValue("@documentTypeId", docTypeId);
                    cmd.ExecuteNonQuery();
                    tablerow = m_tblDocumentTypes.Select("documentTypeId=" + docTypeId)[0];
                    tablerow["docTypeName"] = docType;
                    rows = m_tblDocuments.Select("docTypeId=" + docTypeId);
                    foreach (DataRow dr in rows)
                    {
                        dr["docTypeName"] = docType;
                    }
                }

                m_tblDocumentTypes.AcceptChanges();
                m_tblDocuments.AcceptChanges();

                cn.Close();
            }
        }

        public void SaveDocument(int documentId, int docTypeId, string fileName = "", string fileType = "", DateTime? fileLastEdited = null, long fileSize = 0)
        {
            DataRow tablerow;

            DateTime jetzt = DateTime.Now;

            using (SqlConnection cn = new SqlConnection(m_objApp.Connectionstring))
            {
                cn.Open();

                SqlCommand cmd = new SqlCommand();
                cmd.Connection = cn;
                cmd.CommandType = CommandType.Text;

                if (documentId == -1)
                {
                    // prüfen, ob die Datei schon existiert, also nur überschrieben werden soll, oder wirklich DB-seitig neu angelegt werden muss
                    cmd.CommandText = "SELECT documentId FROM Document WHERE customerId = @CustomerId AND fileName = @fileName AND fileType = @fileType;";
                    cmd.Parameters.AddWithValue("@customerId", m_objUser.Customer.CustomerId);
                    cmd.Parameters.AddWithValue("@fileName", fileName);
                    cmd.Parameters.AddWithValue("@fileType", fileType);
                    object gefundeneId = cmd.ExecuteScalar();

                    if (gefundeneId != null)
                    {
                        cmd.CommandText = "UPDATE Document SET uploaded = @uploaded WHERE documentId = @documentId;";
                        cmd.Parameters.Clear();
                        cmd.Parameters.AddWithValue("@uploaded", jetzt);
                        cmd.Parameters.AddWithValue("@documentId", Convert.ToInt32(gefundeneId));
                        cmd.ExecuteNonQuery();
                        tablerow = m_tblDocuments.Select("documentId='" + Convert.ToInt32(gefundeneId) + "'")[0];
                        tablerow["lastEdited"] = jetzt;
                    }
                    else
                    {
                        cmd.CommandText = "INSERT INTO Document (docTypeId,customerId,fileName,fileType,lastEdited,fileSize,uploaded) VALUES (@docTypeId,@customerId,@fileName,@fileType,@lastEdited,@fileSize,@uploaded);SELECT SCOPE_IDENTITY() AS 'Identity'";
                        cmd.Parameters.Clear();
                        cmd.Parameters.AddWithValue("@docTypeId", docTypeId);
                        cmd.Parameters.AddWithValue("@customerId", m_objUser.Customer.CustomerId);
                        cmd.Parameters.AddWithValue("@fileName", fileName);
                        cmd.Parameters.AddWithValue("@fileType", fileType);
                        cmd.Parameters.AddWithValue("@lastEdited", fileLastEdited);
                        cmd.Parameters.AddWithValue("@fileSize", fileSize);
                        cmd.Parameters.AddWithValue("@uploaded", jetzt);
                        int newId = Convert.ToInt32(cmd.ExecuteScalar());
                        tablerow = m_tblDocuments.NewRow();
                        tablerow["documentId"] = newId;
                        tablerow["docTypeId"] = docTypeId;
                        tablerow["docTypeName"] = m_tblDocumentTypes.Select("documentTypeId=" + docTypeId)[0]["docTypeName"];
                        tablerow["customerId"] = m_objUser.Customer.CustomerId;
                        tablerow["fileName"] = fileName;
                        tablerow["fileType"] = fileType;
                        tablerow["lastEdited"] = fileLastEdited;
                        m_tblDocuments.Rows.Add(tablerow);
                    }
                }
                else
                {
                    // aktuell nur Dokumententyp als veränderbarer Parameter vorgesehen
                    cmd.CommandText = "UPDATE Document SET docTypeId = @docTypeId WHERE documentId = @documentId;";
                    cmd.Parameters.AddWithValue("@docTypeId", docTypeId);
                    cmd.Parameters.AddWithValue("@documentId", documentId);
                    cmd.ExecuteNonQuery();
                    tablerow = m_tblDocuments.Select("documentId='" + documentId + "'")[0];
                    tablerow["docTypeId"] = docTypeId;
                    tablerow["docTypeName"] = m_tblDocumentTypes.Select("documentTypeId=" + docTypeId)[0]["docTypeName"];
                }

                m_tblDocuments.AcceptChanges();

                cn.Close();
            }
        }
    
        public void SaveDocumentRights(int documentId, List<int> groups)
        {
            DataRow tablerow;

            if ((groups != null) && (groups.Count > 0))
            {
                using (SqlConnection cn = new SqlConnection(m_objApp.Connectionstring))
                {
                    cn.Open();

                    SqlCommand cmd = new SqlCommand();
                    cmd.Connection = cn;
                    cmd.CommandType = CommandType.Text;

                    cmd.CommandText = "INSERT INTO DocumentRights (documentId,groupId) VALUES (@documentId,@groupId);";

                    foreach (int gruppe in groups)
                    {
                        cmd.Parameters.Clear();
                        cmd.Parameters.AddWithValue("@documentId", documentId);
                        cmd.Parameters.AddWithValue("@groupId", gruppe);
                        cmd.ExecuteNonQuery();
                        tablerow = m_tblDocumentRights.NewRow();
                        tablerow["documentId"] = documentId;
                        tablerow["groupId"] = gruppe;
                        m_tblDocumentRights.Rows.Add(tablerow);
                    }

                    m_tblDocumentRights.AcceptChanges();

                    cn.Close();
                }
            }
        }

        public bool DeleteDocumentType(int docTypeId)
        {
            DataRow tablerow;

            using (SqlConnection cn = new SqlConnection(m_objApp.Connectionstring))
            {
                cn.Open();

                SqlCommand cmd = new SqlCommand();
                cmd.Connection = cn;
                cmd.CommandType = CommandType.Text;

                // prüfen, ob DocType noch verwendet wird -> dann nicht löschen
                cmd.CommandText = "SELECT COUNT(*) FROM Document WHERE DocTypeId=" + docTypeId + " ;";
                if (Convert.ToInt32(cmd.ExecuteScalar()) > 0)
                {
                    return false;
                }

                cmd.CommandText = "DELETE FROM DocumentType WHERE DocumentTypeId=" + docTypeId + " ;";
                cmd.Parameters.Clear();
                cmd.ExecuteNonQuery();
                tablerow = m_tblDocumentTypes.Select("documentTypeId='" + docTypeId + "'")[0];
                tablerow.Delete();
                m_tblDocumentTypes.AcceptChanges();

                cn.Close();
            }

            return true;
        }

        public void DeleteDocument(int documentId)
        {
            DataRow tablerow;
            DataRow[] rows;

            using (SqlConnection cn = new SqlConnection(m_objApp.Connectionstring))
            {
                cn.Open();

                SqlCommand cmd = new SqlCommand();
                cmd.Connection = cn;
                cmd.CommandType = CommandType.Text;

                // ggf. vorhandene Verknüpfungen entfernen
                cmd.CommandText = "DELETE FROM DocumentRights WHERE DocumentId=" + documentId + " ;";
                cmd.ExecuteNonQuery();
                if ((m_tblDocumentRights != null) && (m_tblDocumentRights.Rows.Count > 0))
                {
                    rows = m_tblDocumentRights.Select("documentId='" + documentId + "'");
                    foreach (DataRow dr in rows)
                    {
                        dr.Delete();
                    }
                    m_tblDocumentRights.AcceptChanges();
                }

                cmd.CommandText = "DELETE FROM Document WHERE DocumentId=" + documentId + " ;";
                cmd.Parameters.Clear();
                cmd.ExecuteNonQuery();
                tablerow = m_tblDocuments.Select("documentId='" + documentId + "'")[0];
                tablerow.Delete();
                m_tblDocuments.AcceptChanges();

                cn.Close();
            }
        }

        public void DeleteDocumentRights(int documentId, List<int> groups)
        {
            DataRow tablerow;

            if ((groups != null) && (groups.Count > 0))
            {
                using (SqlConnection cn = new SqlConnection(m_objApp.Connectionstring))
                {
                    cn.Open();

                    SqlCommand cmd = new SqlCommand();
                    cmd.Connection = cn;
                    cmd.CommandType = CommandType.Text;

                    cmd.CommandText = "DELETE FROM DocumentRights WHERE documentId = @documentId AND groupId = @groupId;";

                    foreach (int gruppe in groups)
                    {
                        cmd.Parameters.Clear();
                        cmd.Parameters.AddWithValue("@documentId", documentId);
                        cmd.Parameters.AddWithValue("@groupId", gruppe);
                        cmd.ExecuteNonQuery();
                        tablerow = m_tblDocumentRights.Select("documentId='" + documentId + "' AND groupId='" + gruppe + "'")[0];
                        tablerow.Delete();
                    }

                    m_tblDocumentRights.AcceptChanges();

                    cn.Close();
                }
            }
        }

        public override void Show()
        {
        }

        public override void Change()
        {
        }

        #endregion

    }
}

File: inetpub/wwwroot/KBS/Lib/InfoCenterData.vb
Option Explicit On
Option Strict On

Imports System
Imports System.Collections.Generic

Public Class InfoCenterData

#Region "Declarations"
    Private m_tblCustomer As New DataTable
    Private m_tblDocumentTypes As New DataTable
    Private m_tblDocuments As New DataTable
    Private m_tblDocumentRights As New DataTable
    Private m_uploadfile As String
#End Region

#Region "Properties"
    Public Property Customer() As DataTable
        Get
            Return m_tblCustomer
        End Get
        Set(ByVal Value As DataTable)
            m_tblCustomer = Value
        End Set
    End Property

    Public Property DocumentTypes() As DataTable
        Get
            Return m_tblDocumentTypes
        End Get
        Set(ByVal Value As DataTable)
            m_tblDocumentTypes = Value
        End Set
    End Property

    Public Property Documents() As DataTable
        Get
            Return m_tblDocuments
        End Get
        Set(ByVal Value As DataTable)
            m_tblDocuments = Value
        End Set
    End Property

    Public Property DocumentRights() As DataTable
        Get
            Return m_tblDocumentRights
        End Get
        Set(ByVal Value As DataTable)
            m_tblDocumentRights = Value
        End Set
    End Property

    Public Property UploadFile As String
        Get
            Return m_uploadfile
        End Get
        Set(value As String)
            m_uploadfile = value
        End Set
    End Property

#End Region

#Region "Methods"

    Public Sub GetCustomerList()
        Using cn As New SqlClient.SqlConnection(ConfigurationManager.AppSettings("Connectionstring"))
            With cn
                .Open()

                m_tblCustomer.Clear()

                Dim daDocTypes As SqlClient.SqlDataAdapter
                daDocTypes = New SqlClient.SqlDataAdapter("SELECT CustomerID, Customername, KUNNR FROM Customer", cn)

                daDocTypes.Fill(m_tblCustomer)

                .Close()
            End With
        End Using
    End Sub

    Public Sub GetDocumentTypes()
        Using cn As New SqlClient.SqlConnection(ConfigurationManager.AppSettings("Connectionstring"))
            With cn
                .Open()

                m_tblDocumentTypes.Clear()

                Dim daDocTypes As SqlClient.SqlDataAdapter
                daDocTypes = New SqlClient.SqlDataAdapter("SELECT * FROM DocumentType", cn)

                daDocTypes.Fill(m_tblDocumentTypes)

                .Close()
            End With
        End Using
    End Sub

    ''' <summary>
    ''' Gibt alle Dokumente bzw. alle Dokumente für einen Kunden zurück
    ''' </summary>
    ''' <param name="customerId"></param>
    ''' <remarks></remarks>
    Public Sub GetDocuments(Optional ByVal customerId As Integer = -1)
        Using cn As New SqlClient.SqlConnection(ConfigurationManager.AppSettings("Connectionstring"))
            With cn
                .Open()

                m_tblDocuments.Clear()

                Dim daDocs As SqlClient.SqlDataAdapter

                If customerId = -1 Then
                    daDocs = New SqlClient.SqlDataAdapter("SELECT * FROM vwDocumentWithType", cn)
                Else
                    daDocs = New SqlClient.SqlDataAdapter("SELECT * FROM vwDocumentCustomer WHERE customerId = @customerId", cn)
                    daDocs.SelectCommand.Parameters.AddWithValue("@customerId", customerId)
                End If

                daDocs.Fill(m_tblDocuments)

                .Close()
            End With
        End Using
    End Sub

    Public Sub GetDocumentRights(ByVal documentId As Integer)
        Using cn As New SqlClient.SqlConnection(ConfigurationManager.AppSettings("Connectionstring"))
            With cn
                .Open()

                m_tblDocumentRights.Clear()

                Dim daDocs As SqlClient.SqlDataAdapter
                daDocs = New SqlClient.SqlDataAdapter("SELECT * FROM documentRights WHERE documentId = @documentId", cn)

                daDocs.SelectCommand.Parameters.AddWithValue("@documentId", documentId)
                daDocs.Fill(m_tblDocumentRights)

                .Close()
            End With
        End Using
    End Sub

    Public Sub SaveDocumentType(ByVal docTypeId As Integer, ByVal docType As String)
        Dim tablerow As DataRow
        Dim rows As DataRow()

        Using cn As New SqlClient.SqlConnection(ConfigurationManager.AppSettings("Connectionstring"))
            With cn
                .Open()

                Dim cmd As SqlClient.SqlCommand = New SqlClient.SqlCommand()
                cmd.Connection = cn
                cmd.CommandType = CommandType.Text

                If docTypeId = -1 Then
                    cmd.CommandText = "INSERT INTO DocumentType (docTypeName) VALUES (@docTypeName);SELECT SCOPE_IDENTITY() AS 'Identity'"
                    cmd.Parameters.AddWithValue("@docTypeName", docType)
                    Dim newId As Integer = CInt(cmd.ExecuteScalar)
                    tablerow = m_tblDocumentTypes.NewRow
                    tablerow("documentTypeId") = newId
                    tablerow("docTypeName") = docType
                    m_tblDocumentTypes.Rows.Add(tablerow)
                Else
                    cmd.CommandText = "UPDATE DocumentType SET docTypeName = @docTypeName WHERE documentTypeId = @documentTypeId;"
                    cmd.Parameters.AddWithValue("@docTypeName", docType)
                    cmd.Parameters.AddWithValue("@documentTypeId", docTypeId)
                    cmd.ExecuteNonQuery()
                    tablerow = m_tblDocumentTypes.Select("documentTypeId=" & docTypeId)(0)
                    tablerow("docTypeName") = docType
                    rows = m_tblDocuments.Select("docTypeId=" & docTypeId)
                    For Each dr As DataRow In rows
                        dr("docTypeName") = docType
                    Next
                End If

                m_tblDocumentTypes.AcceptChanges()
                m_tblDocuments.AcceptChanges()

                .Close()
            End With
        End Using

    End Sub

    Public Sub SaveDocument(ByVal documentId As Integer, ByVal docTypeId As Integer, Optional ByVal fileName As String = "", Optional ByVal fileType As String = "", Optional ByVal fileLastEdited As DateTime = Nothing, Optional ByVal fileSize As Long = 0)
        Dim tablerow As DataRow

        Dim jetzt As DateTime = DateTime.Now

        Using cn As New SqlClient.SqlConnection(ConfigurationManager.AppSettings("Connectionstring"))
            With cn
                .Open()

                Dim cmd As SqlClient.SqlCommand = New SqlClient.SqlCommand()
                cmd.Connection = cn
                cmd.CommandType = CommandType.Text

                If documentId = -1 Then
                    'prüfen, ob die Datei schon existiert, also nur überschrieben werden soll, oder wirklich DB-seitig neu angelegt werden muss
                    cmd.CommandText = "SELECT documentId FROM Document WHERE fileName = @fileName AND fileType = @fileType;"
                    cmd.Parameters.AddWithValue("@fileName", fileName)
                    cmd.Parameters.AddWithValue("@fileType", fileType)
                    Dim gefundeneId As Object = cmd.ExecuteScalar()

                    If gefundeneId IsNot Nothing Then
                        cmd.CommandText = "UPDATE Document SET uploaded = @uploaded WHERE documentId = @documentId;"
                        cmd.Parameters.Clear()
                        cmd.Parameters.AddWithValue("@uploaded", jetzt)
                        cmd.Parameters.AddWithValue("@documentId", CInt(gefundeneId))
                        cmd.ExecuteNonQuery()
                        tablerow = m_tblDocuments.Select("documentId='" & CInt(gefundeneId) & "'")(0)
                        tablerow("lastEdited") = jetzt
                    Else
                        cmd.CommandText = "INSERT INTO Document (docTypeId,fileName,fileType,lastEdited,fileSize,uploaded) VALUES (@docTypeId,@fileName,@fileType,@lastEdited,@fileSize,@uploaded);SELECT SCOPE_IDENTITY() AS 'Identity'"
                        cmd.Parameters.Clear()
                        cmd.Parameters.AddWithValue("@docTypeId", docTypeId)
                        cmd.Parameters.AddWithValue("@fileName", fileName)
                        cmd.Parameters.AddWithValue("@fileType", fileType)
                        cmd.Parameters.AddWithValue("@lastEdited", fileLastEdited)
                        cmd.Parameters.AddWithValue("@fileSize", fileSize)
                        cmd.Parameters.AddWithValue("@uploaded", jetzt)
                        Dim newId As Integer = CInt(cmd.ExecuteScalar)
                        tablerow = m_tblDocuments.NewRow
                        tablerow("documentId") = newId
                        tablerow("docTypeId") = docTypeId
                        tablerow("docTypeName") = m_tblDocumentTypes.Select("documentTypeId=" & docTypeId)(0)("docTypeName")
                        tablerow("fileName") = fileName
                        tablerow("fileType") = fileType
                        tablerow("lastEdited") = fileLastEdited
                        m_tblDocuments.Rows.Add(tablerow)
                    End If
                Else
                    'aktuell nur Dokumententyp als veränderbarer Parameter vorgesehen
                    cmd.CommandText = "UPDATE Document SET docTypeId = @docTypeId WHERE documentId = @documentId;"
                    cmd.Parameters.AddWithValue("@docTypeId", docTypeId)
                    cmd.Parameters.AddWithValue("@documentId", documentId)
                    cmd.ExecuteNonQuery()
                    tablerow = m_tblDocuments.Select("documentId='" & documentId & "'")(0)
                    tablerow("docTypeId") = docTypeId
                    tablerow("docTypeName") = m_tblDocumentTypes.Select("documentTypeId=" & docTypeId)(0)("docTypeName")
                End If

                m_tblDocuments.AcceptChanges()

                .Close()
            End With
        End Using

    End Sub

    Public Sub SaveDocumentRights(ByVal documentId As Integer, ByVal customerList As List(Of Integer))
        Dim tablerow As DataRow

        If customerList IsNot Nothing AndAlso customerList.Count > 0 Then
            Using cn As New SqlClient.SqlConnection(ConfigurationManager.AppSettings("Connectionstring"))
                With cn
                    .Open()

                    Dim cmd As SqlClient.SqlCommand = New SqlClient.SqlCommand()
                    cmd.Connection = cn
                    cmd.CommandType = CommandType.Text

                    cmd.CommandText = "INSERT INTO DocumentRights (documentId,customerId) VALUES (@documentId,@customerId);"

                    For Each kunde As Integer In customerList
                        cmd.Parameters.Clear()
                        cmd.Parameters.AddWithValue("@documentId", documentId)
                        cmd.Parameters.AddWithValue("@customerId", kunde)
                        cmd.ExecuteNonQuery()
                        tablerow = m_tblDocumentRights.NewRow
                        tablerow("documentId") = documentId
                        tablerow("customerId") = kunde
                        m_tblDocumentRights.Rows.Add(tablerow)
                    Next

                    m_tblDocumentRights.AcceptChanges()

                    .Close()
                End With
            End Using
        End If

    End Sub

    Public Function DeleteDocumentType(ByVal docTypeId As Integer) As Boolean
        Dim tablerow As DataRow

        Using cn As New SqlClient.SqlConnection(ConfigurationManager.AppSettings("Connectionstring"))
            With cn
                .Open()

                Dim cmd As SqlClient.SqlCommand = New SqlClient.SqlCommand()
                cmd.Connection = cn
                cmd.CommandType = CommandType.Text

                'prüfen, ob DocType noch verwendet wird -> dann nicht löschen
                cmd.CommandText = "SELECT COUNT(*) FROM Document WHERE DocTypeId=" & docTypeId & " ;"
                If CInt(cmd.ExecuteScalar()) > 0 Then
                    Return False
                End If

                cmd.CommandText = "DELETE FROM DocumentType WHERE DocumentTypeId=" & docTypeId & " ;"
                cmd.Parameters.Clear()
                cmd.ExecuteNonQuery()
                tablerow = m_tblDocumentTypes.Select("documentTypeId='" & docTypeId & "'")(0)
                tablerow.Delete()
                m_tblDocumentTypes.AcceptChanges()

                .Close()
            End With
        End Using

        Return True

    End Function

    Public Sub DeleteDocument(ByVal documentId As Integer)
        Dim tablerow As DataRow
        Dim rows As DataRow()

        Using cn As New SqlClient.SqlConnection(ConfigurationManager.AppSettings("Connectionstring"))
            With cn
                .Open()

                Dim cmd As SqlClient.SqlCommand = New SqlClient.SqlCommand()
                cmd.Connection = cn
                cmd.CommandType = CommandType.Text

                'ggf. vorhandene Verknüpfungen entfernen
                cmd.CommandText = "DELETE FROM DocumentRights WHERE DocumentId=" & documentId & " ;"
                cmd.ExecuteNonQuery()
                If m_tblDocumentRights IsNot Nothing AndAlso m_tblDocumentRights.Rows.Count > 0 Then
                    rows = m_tblDocumentRights.Select("documentId='" & documentId & "'")
                    For Each dr As DataRow In rows
                        dr.Delete()
                    Next
                    m_tblDocumentRights.AcceptChanges()
                End If

                cmd.CommandText = "DELETE FROM Document WHERE DocumentId=" & documentId & " ;"
                cmd.Parameters.Clear()
                cmd.ExecuteNonQuery()
                tablerow = m_tblDocuments.Select("documentId='" & documentId & "'")(0)
                tablerow.Delete()
                m_tblDocuments.AcceptChanges()

                .Close()
            End With
        End Using

    End Sub

    Public Sub DeleteDocumentRights(ByVal documentId As Integer, ByVal customerList As List(Of Integer))
        Dim tablerow As DataRow

        If customerList IsNot Nothing AndAlso customerList.Count > 0 Then
            Using cn As New SqlClient.SqlConnection(ConfigurationManager.AppSettings("Connectionstring"))
                With cn
                    .Open()

                    Dim cmd As SqlClient.SqlCommand = New SqlClient.SqlCommand()
                    cmd.Connection = cn
                    cmd.CommandType = CommandType.Text

                    cmd.CommandText = "DELETE FROM DocumentRights WHERE documentId = @documentId AND customerId = @customerId;"

                    For Each kunde As Integer In customerList
                        cmd.Parameters.Clear()
                        cmd.Parameters.AddWithValue("@documentId", documentId)
                        cmd.Parameters.AddWithValue("@customerId", kunde)
                        cmd.ExecuteNonQuery()
                        tablerow = m_tblDocumentRights.Select("documentId='" & documentId & "' AND customerId='" & kunde & "'")(0)
                        tablerow.Delete()
                    Next

                    m_tblDocumentRights.AcceptChanges()

                    .Close()
                End With
            End Using
        End If

    End Sub

#End Region

End Class

